internal/clients/diffreviewer/circleci/circleci_service_test.go

```diff
@@ -0,0 +1,97 @@
+package circleci
+
+import (
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	circlelogger "github.com/nightfallai/nightfall_code_scanner/internal/clients/logger/circle_logger"
+	"github.com/nightfallai/nightfall_code_scanner/internal/nightfallconfig"
+	nightfallAPI "github.com/nightfallai/nightfall_go_client/generated"
+	"github.com/stretchr/testify/suite"
+)
+
+var circleLogger = circlelogger.NewDefaultCircleLogger()
+
+type circleCiTestSuite struct {
+	suite.Suite
+}
+
+type testParams struct {
+	ctrl *gomock.Controller
+	cs   *Service
+	w    *httptest.ResponseRecorder
+}
+
+func (c *circleCiTestSuite) initTestParams() *testParams {
+	tp := &testParams{}
+	tp.ctrl = gomock.NewController(c.T())
+	tp.w = httptest.NewRecorder()
+	tp.cs = &Service{
+		Logger: circleLogger,
+	}
+	return tp
+}
+
+const commitSha = "7b46da6e4d3259b1a1c470ee468e2cb3d9733802"
+const prevCommitSha = "15bf9548d16caff9f398b5aae78a611fc60d55bd"
+const testConfigFileName = "nightfall_test_config.json"
+const excludedCreditCardRegex = "4242-4242-4242-[0-9]{4}"
+const excludedApiToken = "xG0Ct4Wsu3OTcJnE1dFLAQfRgL6b8tIv"
+const excludedIPRegex = "^127\\."
+
+var envVars = []string{
+	WorkspacePathEnvVar,
+	NightfallAPIKeyEnvVar,
+	CircleCurrentCommitShaEnvVar,
+	CircleBeforeCommitEnvVar,
+}
+
+func (c *circleCiTestSuite) AfterTest(suiteName, testName string) {
+	for _, e := range envVars {
+		err := os.Unsetenv(e)
+		c.NoErrorf(err, "Error unsetting var %s", e)
+	}
+}
+
+func (c *circleCiTestSuite) TestLoadConfig() {
```

Comment:
since we have the weirdness w/ the Env variables and checking their existence in the LoadConfig method, lets add a test or two to make sure that we return the correct error if they're empty. We don't need to check all of them but at least the NIGHTFALL_API_KEY